plugins/qeta/src/components/MarkdownEditor/MarkdownEditor.tsx

```diff
@@ -1,23 +1,64 @@
+/* eslint-disable no-console */
 import React from 'react';
 import ReactMde from 'react-mde';
+
+import { Config } from '@backstage/config';
 import { MarkdownContent } from '@backstage/core-components';
 import 'react-mde/lib/styles/css/react-mde.css';
 import 'react-mde/lib/styles/css/react-mde-editor.css';
 import 'react-mde/lib/styles/css/react-mde-toolbar.css';
 import { useStyles } from '../../utils/hooks';
 
+export const submitImage = async (config: Config, file: Blob) => {
+  const backendBaseUrl = config.getString('backend.baseUrl');
+  const qetaUrl = `${backendBaseUrl}/api/qeta/attachments`;
+  const formData = new FormData();
+
+  formData.append('image', file);
+
+  const requestOptions = {
+    method: 'POST',
+    body: formData,
+  };
+
+  const response = await fetch(qetaUrl, requestOptions);
+
+  if (response.status >= 400) {
+    const responseError = await response.text();
+    throw new Error(`Failed to upload image question : ${responseError}`);
+  }
+  return response.json();
+};
+
+const imageUpload = (config: Config) => {
+  // eslint-disable-next-line func-names
+  return async function* (data: ArrayBuffer) {
+    const { imageURL } = await submitImage(
+      config,
+      new Blob([data], { type: 'text/plain' }),
```

Comment:
You need to figure out the proper mime type for this in the upload phase. I think users should also allow selecting which kind of files they allow the users to upload in configuration. Now they can basically upload anything. Also, file size limit would be a nice addition to prevent misuse with some large files.